### PR TITLE
feat: early stopping 

### PR DESCRIPTION
**Pull request by erabinov**

### CODE REVIEW

File: fri/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //! An implementation of the FRI low-degree test (LDT).
 
-#![no_std]
+// #![no_std]
```
Comment (erabinov) [inline fri/src/lib.rs:3]:
```suggestion
#![no_std]
```

File: fri/src/prover.rs [hunk 2]
```diff
@@ -25,10 +27,17 @@ where
...
+
     let commit_phase_result = commit_phase(config, input, log_max_height, challenger);
 
     let pow_witness = challenger.grind(config.proof_of_work_bits);
 
+    // input.iter().enumerate().for_each(|(i, v)| {
+    //     if let Some(_) = v {
+    //     }
+    // });
+
```
Comment (erabinov) [inline fri/src/prover.rs:40]:
```suggestion
```

File: fri/src/prover.rs [hunk 3]
```diff
@@ -120,12 +129,19 @@ where
...
-    for x in current {
-        assert_eq!(x, final_poly);
-    }
-    challenger.observe_ext_element(final_poly);
+    // assert_eq!(current.len(), config.blowup());
+    // let final_poly = current[0];
+    // for x in current {
+    //     assert_eq!(x, final_poly);
+    // })
+
```
Comment (erabinov) [inline fri/src/prover.rs:137]:
```suggestion
```

File: fri/src/lib.rs (hunk 1 above)
Comment (jtguibas) [inline fri/src/lib.rs:3]: @erabinov you should fix this!

File: fri/src/verifier.rs [hunk 4]
```diff
@@ -154,8 +173,15 @@ where
         x = x.square();
     }
 
-    debug_assert!(index < config.blowup(), "index was {}", index);
-    debug_assert_eq!(x.exp_power_of_2(config.log_blowup), F::one());
+    // debug_assert!(
```
Comment (jtguibas) [inline fri/src/verifier.rs:176]: nit: clean this up

**Pull request closed without merging**